pinot-common/src/main/java/org/apache/pinot/common/request/context/RequestContextUtils.java

```diff
@@ -297,12 +300,18 @@ public static FilterContext getFilter(FunctionContext filterFunction) {
         for (int i = 1; i < numOperands; i++) {
           values.add(getStringValue(operands.get(i)));
         }
+        if (numOperands > THRESHOLD_OF_LARGE_IN_CLAUSE_VALUES) {
+          Collections.sort(values);
```

Comment:
I'm not sure this is valid unless the column type is `STRING`

Reply:
Also, take a look at galloping intersection here https://github.com/RoaringBitmap/RoaringBitmap/blob/fd54c0a100629bb578946e2a0bf8b62784878fa8/RoaringBitmap/src/main/java/org/roaringbitmap/Util.java#L1028